Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+        """
+        return self.connected
+
+    def is_armed(self):
+        """
+        Getter for self.vehicle.armed
+
+        Parameters
+        ----------
+        None
+
+        Returns:
+        ----------
+        Boolean
+            True if the drone is armed and false otherwise
+        """
+        return self.vehicle.armed
+
+    def takeoff(self, target_altitude, stop_event):
+        """
+        The drone flies from the ground to desired altitude
+
+        Parameters
+        ----------
+        target_altitude: Integer 
+            The height in meters the drone should be off the ground after
+            this function completes
+        stop_event: threading.Event
+            Event which is set when takeoff should be canceled
+
+        Precondition:
+        ----------
+        The drone is connected and armed.
+
+        Postcondition:
+        ----------
+        Upon success: the drone is the requested altitude in the air. 
+
+        Returns:
+        ----------
+        None
+        """
+        def takeoff_thread(target_altitude, finished, stop_event):
+            self.logger.info(threading.current_thread().name + ": Starting takeoff")
+            thrust = c.DEFAULT_TAKEOFF_THRUST
+
+            start_time = time.time()
+            cutoff_time = 10
+            
+            while time.time() - start_time < cutoff_time:
+                if stop_event.is_set():
+                    self.logger.info(threading.current_thread().name + ": Takeoff halting")
+                    break
+
+                current_altitude = self.altitude()
+
+                if current_altitude >= target_altitude*0.95: # Trigger just below target alt.
+                    break
+                elif current_altitude >= target_altitude*0.6:
+                    thrust = c.SMOOTH_TAKEOFF_THRUST
+
+                dkw.set_attitude(self.vehicle, thrust=thrust)
+                time.sleep(1)
+            else:
+                self.logger.error("Could not take off!")
+            
+            self.logger.info(threading.current_thread().name + ": Finished takeoff")
+            finished.set()
+
+        finished = threading.Event()
+        threading.Thread(target=takeoff_thread, name="TakeoffThread-" + str(DroneBase.cnt), 
+            args=(target_altitude, finished, stop_event)).start()
+        DroneBase.cnt += 1
+        return finished
+
+    def land(self):
+        """
+        Lands the drone on the ground
+
+        Parameters
+        ----------
+        None
+
+        Precondition:
+        ----------
+        The drone is connected and armed.
+
+        Postcondition:
+        ----------
+        The drone is on the groun (and ideally not smashed into 1000 pieces)
+        
+        Returns:
+        ----------
+        None
+        """
+        def land_thread(finished):
+            self.logger.info(threading.current_thread().name + ": Starting land")
+            while not self.vehicle.mode == VehicleMode(c.LAND_MODE):
```

Comment:
Code should follow the DRY principle. Set an intermediate variable and continue on as before 
```suggestion
            land_mode = VehicleMode(c.LAND_MODE)
            while not self.vehicle.mode == land_mode:
```